dashboard/test/controllers/api_controller_test.rb

```diff
@@ -1141,6 +1157,70 @@ def get_student_response(script, level, lesson, student_number)
     assert_equal 1, response["script"]["lessons"][0]["length"]
   end
 
+  test "teacher_panel_progress returns progress when called with script and level" do
+    script, _, regular_level, _ = create_script_with_bonus_levels
+
+    # create progress for student_1 on regular_level
+    create :user_level, user: @student_1, script: script, level: regular_level, best_result: ActivityConstants::BEST_PASS_RESULT
+
+    get :teacher_panel_progress, params: {
+      section_id: @section.id,
+      script_id: script.id,
+      level_id: regular_level.id
+    }
+
+    assert_response :success
+
+    response = JSON.parse(@response.body)
+
+    # response is an array with one element for each student
+    # students are sorted by name so @student_1 should be the first result
+    assert_equal @students.length, response.length
+    assert_equal @student_1.id, response[0]["userId"]
+    assert_equal regular_level.id.to_s, response[0]["id"]
+    assert_equal "perfect", response[0]["status"]
+  end
+
+  test "teacher_panel_progress returns progress when called with lesson and is_bonus_lesson" do
+    script, lesson, _, bonus_level = create_script_with_bonus_levels
+
+    # create progress for student_1 on bonus_level
+    create :user_level, user: @student_1, script: script, level: bonus_level, best_result: ActivityConstants::BEST_PASS_RESULT
+
+    get :teacher_panel_progress, params: {
+      section_id: @section.id,
+      script_id: script.id,
+      lesson_id: lesson.id,
+      is_lesson_extras: true
+    }
+
+    assert_response :success
+
+    response = JSON.parse(@response.body)
+
+    # response is an array with one element for each student
+    # students are sorted by name so @student_1 should be the first result
+    assert_equal @students.length, response.length
+    assert_equal @student_1.id, response[0]["userId"]
+    assert_equal bonus_level.id.to_s, response[0]["id"]
+    assert_equal "perfect", response[0]["status"]
+  end
+
+  test "teacher_panel_progress returns error when called by teacher not associated with section" do
+    sign_in @teacher_other
+
+    get :teacher_panel_progress, params: {
+      section_id: @section.id,
+      script_id: @script.id
+    }
+
+    assert_response :forbidden
+  end
+
+  # test "" do
```

Comment:
Missing test?

Reply:
Would you mind just removing this test for me so we can avoid having to coordinate commits?